metrics-publishers/cloudwatch-metrics-publisher/src/test/java/software/amazon/awssdk/metrics/publishers/cloudwatch/CloudWatchPublisherIntegrationTest.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright 2010-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.metrics.publishers.cloudwatch;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.atLeast;
+
+import java.time.Duration;
+import java.util.List;
+import java.util.stream.IntStream;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.runners.MockitoJUnitRunner;
+import software.amazon.awssdk.metrics.meter.Counter;
+import software.amazon.awssdk.metrics.meter.DefaultGauge;
+import software.amazon.awssdk.metrics.meter.Gauge;
+import software.amazon.awssdk.metrics.meter.LongCounter;
+import software.amazon.awssdk.metrics.registry.DefaultMetricRegistry;
+import software.amazon.awssdk.metrics.registry.MetricRegistry;
+import software.amazon.awssdk.services.cloudwatch.CloudWatchAsyncClient;
+import software.amazon.awssdk.services.cloudwatch.model.MetricDatum;
+import software.amazon.awssdk.services.cloudwatch.model.PutMetricDataRequest;
+import software.amazon.awssdk.testutils.service.AwsIntegrationTestBase;
+
+@RunWith(MockitoJUnitRunner.class)
+public class CloudWatchPublisherIntegrationTest extends AwsIntegrationTestBase {
```

Comment:
Can we rename the test since it's not an integration test? Can we add an integration test?

Reply:
Renamed. Will add integration test once I combine this and the other PR.